icu4c/source/i18n/basictz.cpp

```diff
@@ -293,71 +293,77 @@ BasicTimeZone::getTimeZoneRulesAfter(UDate start, InitialTimeZoneRule*& initial,
     }
 
     const InitialTimeZoneRule *orgini;
-    const TimeZoneRule **orgtrs = NULL;
     TimeZoneTransition tzt;
-    UBool avail;
-    UVector *orgRules = NULL;
+    bool avail;
     int32_t ruleCount;
-    TimeZoneRule *r = NULL;
-    UBool *done = NULL;
-    InitialTimeZoneRule *res_initial = NULL;
-    UVector *filteredRules = NULL;
+    TimeZoneRule *r = nullptr;
     UnicodeString name;
     int32_t i;
     UDate time, t;
-    UDate *newTimes = NULL;
     UDate firstStart;
-    UBool bFinalStd = FALSE, bFinalDst = FALSE;
+    UBool bFinalStd = false, bFinalDst = false;
+
+    initial = nullptr;
+    transitionRules = nullptr;
 
     // Original transition rules
     ruleCount = countTransitionRules(status);
     if (U_FAILURE(status)) {
         return;
     }
-    orgRules = new UVector(ruleCount, status);
+    LocalPointer<UVector> orgRules(
+        new UVector(uprv_deleteUObject, nullptr, ruleCount, status), status);
     if (U_FAILURE(status)) {
         return;
     }
-    orgtrs = (const TimeZoneRule**)uprv_malloc(sizeof(TimeZoneRule*)*ruleCount);
-    if (orgtrs == NULL) {
+    LocalMemory<const TimeZoneRule *>orgtrs(
```

Comment:
super nit/optional: The rest of the file seems to use a space between the variable type and the name.
```suggestion
    LocalMemory<const TimeZoneRule *> orgtrs(
```